oemof/solph/models.py

```diff
@@ -96,7 +96,8 @@ def __init__(self, es, **kwargs):
 
         # non-negative pyomo variable for all existing flows in energysystem
         self.flow = po.Var(self.FLOWS, self.TIMESTEPS,
-                           within=po.NonNegativeReals)
+                           within=po.Reals)
```

Comment:
I think these lines will break existing models.

Would it be okay to add a flow option like "NonNegative=True" and you can set it to False to make lopf possible?

Reply:
I added a condition to set the lower bound to zero. Acutally, I think it would be great if generally we allow to be the variables within Reals and then restrict it by bounds (min, max etc). Actually I think it should not have broken model results but only lp-files before

Reply:
> I think it would be great if generally we allow to be the variables within Reals and then restrict it by bounds (min, max etc)

I thought the default value of min is "0" anyway shouldn't it work without your hack? If min=0 is set we do not need my proposal above. But somehow it does not seem to work.

Reply:
yes thats what i meant, but it ll definitely break lp files. the thing is i guess this may be a problem for unrestricted flows, e.g. from a gas source to a gas bus. in that case no bounds are set and the change from NonNegReals to Reals changes the flow bounds for these flows. but in that case one line should also be sufficient : 

`self.flow[o, i, t].setlb(0)`

This would mean: first set all flows lower bound to '0' and then maybe reset if `min` is negative

Reply:
If I understand you correctly this is similar to what I was thinking of this morning 😄 

1. At the moment the default value of min is "0". That means that we could check if any value within the min sequence is below 0. If this is the case we could set the variable to `Reals` otherwise to `NonNegReals`.

2. In the next step we allow components to overwrite the default value of the connected flows. For example the "electrical_line" could loop over all flows and set `min` to -1 if `min` is `None`.

In that case the changes are not somewhere in the code but directly within the definition of the component. I think this is more explicit.

We could of course discuss about what is the default behaviour and what is the exception in the future but by now I would keep as described above.

Reply:
BTW : Breaking the lp-files is no problem for me as long as the the models still work.

Reply:
Not possible to set Reals for some flows and NonReals for some other flows, either the one or the other for all variables. In my opinion we should just do flow as reals and set lower bounds based on flow objects and keep default value of 0 for all flows. we just need to make that work.

Reply:
> Not possible to set Reals for some flows and NonReals for some other flows, either the one or the other for all variables.

Did not know that.

> we just need to make that work.

That's right 😏 Does `self.flow[o, i, t].setlb(min[t])` work? Will this make the problem bigger/slower? In that case we could think of a NonNegativeFlowModel as special child Model.

Reply:
not bigger or slower to solve, maybe some milliseconds slower to construct. 

but of course it would need to be min * nominal_value ...therefore I like the idea that I implemented below: just make the default `0`, but allow for negative values. This would not change anything  (API or models etc) for anyone but allow for more developing flexibility for us